src/context/engagement_context/engagement_context.tsx

```diff
@@ -228,8 +272,16 @@ export const EngagementProvider = ({
         let errorMessage =
           'There was an issue with saving your changes. Please followup with an administrator if this continues.';
         if (e instanceof AlreadyExistsError) {
-          errorMessage =
-            'The path that you input is already taken.  Please update and try saving again.';
+          // If there is no mongo id associated with the engagement, then it is being committed to the backend for the first time. It is a net new engagement.
```

Comment:
@dwasinge Is this logic sound? is there ever a time we would create an engagement and not get a mongo_id back?